tapdb/universe.go

```diff
@@ -617,4 +617,83 @@ func (b *BaseUniverseTree) MintingLeaves(ctx context.Context,
 	return leaves, nil
 }
 
+// DeleteUniverse deletes the entire universe tree.
+func (b *BaseUniverseTree) DeleteUniverse(ctx context.Context) (string, error) {
+	var writeTx BaseUniverseStoreOptions
+
+	dbErr := b.db.ExecTx(ctx, &writeTx, func(db BaseUniverseStore) error {
+		// Fetch all base keys for the current universe.
+		universeKeys, err := db.FetchUniverseKeys(ctx, b.smtNamespace)
```

Comment:
Hmm, if we decide to go with deletion directly on the table, then we can speed things up quite a bit. As is, since we call `Delete` with the key below, it'll actually do all the normal SMT operations (bubble up the new sum, etc, etc) that isn't actually needed.

Reply:
We only want to just wipe all the relate smt nodes from the db. Using the `Delete` call as below will actually recompute the root each time, which causes extra reads/insertions. Instead we can implement the deletion purely here in the db (delete all leaves w/ namespace, then root), side stepping the main external interface.